### ISSUE TITLE
Increase the w-terms corrections factor

### ISSUE THREAD

**Issue by viralp** [caracal-pipeline/caracal#1187]
I could not find `-nwlayers-factor` option in the current selfcal worker. I found w-terms related artefacts so want to increase the default w-layers. Could any one please add it? Thanks.

**Comment by IanHeywood** [comment]
Setting it to 3 by default will make the artefacts go away for MeerKAT L-band work, without too much of a performance hit.

**Comment by KshitijT** [comment]
Thanks, I'll add the options.

**Comment by KshitijT** [comment]
FYI, this needs a stimela parameter by the same name, which it lacks right now..

**Comment by KshitijT** [comment]
@viralp , I have added the option to the `issue-1187` branch (which needs the stimela branch nwlayer_factor to run). Please try it out if it helps.

BTW, looks like wsclean anyway makes the number of w-layers to the available cores; so this problem can also be sorted by not setting the ncpu option in the master branch of caracal on a machine with enough cores; see:

```
# The theoretically suggested number of w-layers (13) is less than the number of availables
# cores (30). Changing suggested number of w-layers to 30.
```
But of course if there are many more w-layers needed that's not going to help. 

Anyway, don't close this issue until #1201 goes through.

**Comment by ianjarog** [comment]
@KshitijT @viralp  So I am having this error now. If I understand, I need to install the stimela branch nwlayer_factor and that's it or do I have to change something on the yml?  If so what is the exact keyword w-layers or -nwlayers-factor?

**Comment by KshitijT** [comment]
@ianjarog, the name of the parameter in _CARACal_ is `img_nwlayers_factor`. You need to install both the stimela branch nwlayer_factor and the CARACal branch `issue-1187` for it to work.

**Comment by viralp** [comment]
@ianjarog I set ```img_nwlayers_factor: 3``` in the selfcal worker. As @KshitijT mentioned I installed the ```issue-1187``` branch for  Caracal and ```nwlayer_factor``` branch for Stimela. I got decent result with ```img_nwlayers_factor: 3``` as you can see below. The low w-terms related arc-like artefacts (left) are disappeared (right).     

![image](https://user-images.githubusercontent.com/4331123/88110825-22b09500-cbad-11ea-8126-d38323a26acb.png)

**Comment by KshitijT** [comment]
Fixed through #1201 .
